### PR TITLE
fix: [CDS-51808]: hosted cluster support added

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx [hunk 1]
```diff
@@ -575,7 +575,7 @@ export const getAppPayload = (props: PayloadInterface) => {
           repoId: repositoryData?.identifier
         },
         destination: {
-          namespace: clusterData?.namespaces,
+          namespace: 'default',
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx:578]: Shall we do clusterData?.namespaces || 'default'?

File: src/modules/75-cd/pages/get-started-with-cd/ConfigureGitops/DestinationStep.tsx [hunk 2]
```diff
@@ -104,6 +105,15 @@ export const DestinationStep = (props: any) => {
     lazy: true
   })
 
+  const { error: hostedClusterError, mutate: createHostedCluster } = useAgentClusterServiceCreateHosted({
+    agentIdentifier: fullAgentName,
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/ConfigureGitops/DestinationStep.tsx:111]: This is done since BE has some issue. @manavjot-harness is fixing it.

File: src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/75-cd/pages/get-started-with-cd/CDOnboardingUtils.tsx:578]: will do

File: src/modules/75-cd/pages/get-started-with-cd/ConfigureGitops/DestinationStep.tsx [hunk 3]
```diff
@@ -270,6 +280,117 @@ export const DestinationStep = (props: any) => {
     }
   }
 
+  const ProvisionCluster = (): React.ReactElement => {
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/ConfigureGitops/DestinationStep.tsx:283]: As discussed we will move this component to separate file in another PR

**Review by ishantmehtaharness — DISMISSED** [review]
Looks Good

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger jest

**Review by ishantmehtaharness — APPROVED** [review]
